File: app/src/main/java/com/earth/OsToolkit/Fragment/DeviceInfoFragment.java
package com.earth.OsToolkit.Fragment;

import android.app.ActivityManager;
import android.content.Context;
import android.os.*;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;
import android.view.*;

import com.earth.OsToolkit.Items.DeviceInfoCardView;

import static com.earth.OsToolkit.Items.DeviceInfoCardView.Item;

import com.earth.OsToolkit.R;

import java.util.Timer;
import java.util.TimerTask;

import static com.earth.OsToolkit.Base.Checking.*;

public class DeviceInfoFragment extends Fragment {

	/*
	 * 27 Dec 2018
	 *
	 * By 1552980358
	 *
	 */

	 FragmentManager fragmentManager;
	 Fragment fragment_loading = new LoadingFragment();

	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		fragmentManager = getActivity().getSupportFragmentManager();

		FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
		fragmentTransaction.add(R.id.main_fragment,fragment_loading).show(fragment_loading).hide(this).commit();
	}

	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater,
	                         @Nullable ViewGroup container,
	                         @Nullable Bundle savedInstanceState) {

		View view = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_deviceinfo,null);

		setGeneral(view);
		setAndroid(view);
		setSoC(view);
		setRam(view);
		setRuntime(view);

		return view;
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		Fragment current = this;

		Timer timer = new Timer();
		timer.schedule(new TimerTask() {
			@Override
			public void run() {
				FragmentTransaction fragmentTransactionEnd = fragmentManager.beginTransaction();
				fragmentTransactionEnd.show(current).remove(fragment_loading).commit();
			}
		},1000);
	}

	public void setGeneral(View view) {
		DeviceInfoCardView general = view.findViewById(R.id.deviceinfo_general);
		general.setTitle(R.string.deviceinfo_general_title);

		Item manufacturer = new Item(getActivity(), R.string.deviceinfo_general_manufacturer,
				Character.toUpperCase(android.os.Build.MANUFACTURER.charAt(0)) +
						android.os.Build.MANUFACTURER.substring(1));
		Item brand = new Item(getActivity(), R.string.deviceinfo_general_brand,
				Character.toUpperCase(android.os.Build.BRAND.charAt(0))
						+ android.os.Build.BRAND.substring(1));
		Item model = new Item(getActivity(), R.string.deviceinfo_general_model, Build.MODEL);

		Item device = new Item(getActivity(), R.string.deviceinfo_general_device, Build.DEVICE);

		general.addItems(manufacturer, brand, model, device);

	}

	public void setAndroid(View view) {
		DeviceInfoCardView android = view.findViewById(R.id.deviceinfo_android);

		android.setTitle(getString(R.string.deviceinfo_android_title));

		Item version = new Item(getActivity(),R.string.deviceinfo_android_version, getAndroidVersion());

		Item sdk = new Item(getActivity(),"SDK", Build.VERSION.SDK_INT + "");

		Item type = new Item(getActivity(),R.string.deviceinfo_android_type, Build.TYPE);

		android.addItems(version,sdk,type);

	}

	public void setSoC(View view) {
		DeviceInfoCardView soc = view.findViewById(R.id.deviceinfo_soc);
		soc.setTitle(R.string.deviceinfo_soc_title);

		Item manufacturer = new Item(getActivity(), R.string.deviceinfo_soc_manufacturer,
				Character.toUpperCase(Build.HARDWARE.charAt(0)) +
						Build.HARDWARE.substring(1));

		Item product = new Item(getActivity(), R.string.deviceinfo_soc_model, Build.BOARD);

		Item cores = new Item(getActivity(), R.string.deviceinfo_soc_cores, getCPUCores() + "");

		Item abi = new Item(getActivity(), R.string.deviceinfo_soc_abis, getAllAPI());

		Item abi64 = new Item(getActivity(), R.string.deviceinfo_soc_abis64, get64API());

		Item abi32 = new Item(getActivity(), R.string.deviceinfo_soc_abis32, get32API());

		soc.addItems(manufacturer, product, cores, abi, abi64, abi32);
	}


	public void setRam(View view) {
		ActivityManager.MemoryInfo memoryInfo = new ActivityManager.MemoryInfo();
		ActivityManager activityManager = (ActivityManager) getActivity().getSystemService(Context.ACTIVITY_SERVICE);
		activityManager.getMemoryInfo(memoryInfo);

		long totalMem = memoryInfo.totalMem;
		Log.i("totalMem", totalMem + "");

		long threshold = memoryInfo.threshold;

		DeviceInfoCardView ram = view.findViewById(R.id.deviceinfo_ram);
		ram.setTitle(R.string.deviceinfo_ram_title);

		Item totalMemory = new Item(getActivity(), R.string.deviceinfo_ram_totalMem, convertUnit(totalMem));

		Item thresholdMem = new Item(getActivity(), R.string.deviceinfo_ram_threshold, convertUnit(threshold));

		ram.addItems(totalMemory, thresholdMem);

	}

	public void setRuntime(View view) {
		DeviceInfoCardView runtime = view.findViewById(R.id.deviceinfo_runtime);
		runtime.setTitle(R.string.deviceinfo_javaruntime_title);

		Runtime r = Runtime.getRuntime();

		long maxMemory = r.maxMemory();
		long totalMemory = r.totalMemory();

		Item maxMem = new Item(getActivity(), R.string.deviceinfo_javaruntime_maxmem, convertUnit(maxMemory));

		Item totalMem = new Item(getActivity(), R.string.deviceinfo_javaruntime_totalmem, convertUnit(totalMemory));

		runtime.addItems(maxMem, totalMem);
	}


	private String convertUnit(long no) {
		if (no > 1024 * 1024) {
			return no / 1024 / 1024 + " MB";
		} else if (no > 1024) {
			return no / 1024 + "KB";
		} else {
			return no + "B";
		}
	}

}

File: app/src/main/java/com/earth/OsToolkit/fragments/DeviceInfoFragment.kt
package com.earth.OsToolkit.fragments

import android.app.Dialog
import android.content.Context
import android.os.*
import android.support.v4.app.Fragment
import android.view.*

import com.earth.OsToolkit.R
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getAvailableCore
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getABI32
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getABI64
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getABIs
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getAndroidVersion
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.getAndroidVersionName
import com.earth.OsToolkit.base.BaseKotlinOperation.Companion.unitConvert
import com.earth.OsToolkit.base.BaseJavaOperation
import com.earth.OsToolkit.view.DeviceInfoView.ChildView

import kotlinx.android.synthetic.main.fragment_deviceinfo.*

import java.util.*

/*
 * OsToolkit - Kotlin
 *
 * Date : 1/1/2019
 *
 * By   : 1552980358
 *
 */

/*
 * Modify
 *
 * 9/1/2019
 * 23/1/2019
 *
 */

//@SuppressWarnings("all")
class DeviceInfoFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_deviceinfo, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val dialog = Dialog(activity as Context)
        dialog.setCancelable(false)
        dialog.setContentView(LayoutInflater.from(activity).inflate(R.layout.dialog_loading, null))
        dialog.show()

        val t1 = Thread {
            val m = ChildView(
                activity!!,
                R.string.deviceinfo_general_manufacturer,
                Build.MANUFACTURER[0].toUpperCase() + Build.MANUFACTURER.substring(1)
            )
            val b = ChildView(
                activity!!,
                R.string.deviceinfo_general_brand,
                Build.BRAND[0].toUpperCase() + Build.BOARD.substring(1)
            )
            val mo = ChildView(activity!!, R.string.deviceinfo_general_model, Build.MODEL)
            val d = ChildView(activity!!, R.string.deviceinfo_general_device, Build.DEVICE)
            val p = ChildView(activity!!, R.string.deviceinfo_general_product, Build.PRODUCT)
            val h = ChildView(activity!!, R.string.deviceinfo_general_hardware, Build.HARDWARE)

            activity!!.runOnUiThread { generalRoot.addViews(m, b, mo, d, p, h) }
        }
        t1.start()

        val t2 = Thread {
            val v = ChildView(activity!!, R.string.deviceinfo_android_version, getAndroidVersion())
            val vN = ChildView(activity, R.string.deviceinfo_android_versionName, getAndroidVersionName())
            val s = ChildView(activity, R.string.deviceinfo_android_sdk, Build.VERSION.SDK_INT.toString())
            val t = ChildView(activity, R.string.deviceinfo_android_type, Build.TYPE)

            activity!!.runOnUiThread { androidRoot.addViews(v, vN, s, t) }
        }
        t2.start()

        val t3 = Thread {
            val b = ChildView(activity, R.string.deviceinfo_soc_board, Build.BOARD)
            val c = ChildView(activity, R.string.deviceinfo_soc_cores, getAvailableCore().toString())
            val aa = ChildView(activity, R.string.deviceinfo_soc_abis, getABIs())
            val a64 = ChildView(activity, R.string.deviceinfo_soc_abis64, getABI64())
            val a32 = ChildView(activity, R.string.deviceinfo_soc_abis32, getABI32())

            activity!!.runOnUiThread { socRoot.addViews(b, c, aa, a64, a32) }
        }

        t3.start()

        val t4 = Thread {
            val memoryInfo = BaseJavaOperation.getMemory(activity)
            val to = ChildView(activity, R.string.deviceinfo_ram_totalMem, unitConvert(memoryInfo.totalMem))
            val th = ChildView(activity, R.string.deviceinfo_ram_threshold, unitConvert(memoryInfo.threshold))

            activity!!.runOnUiThread { ramRoot.addViews(to, th) }
        }
        t4.start()

        val t5 = Thread {
            val runtime = Runtime.getRuntime()

            val m = ChildView(activity, R.string.deviceinfo_jvm_maxmem, unitConvert(runtime.maxMemory()))
            val t = ChildView(activity, R.string.deviceinfo_jvm_totalmem, unitConvert(runtime.totalMemory()))

            activity!!.runOnUiThread { jvmRoot.addViews(m, t) }
        }
        t5.start()

        val t6 = Thread {

            val displayMatrix = activity!!.resources.displayMetrics
            val x = displayMatrix.widthPixels
            val y = displayMatrix.heightPixels

            val r = ChildView(activity, R.string.deviceinfo_display_resolution, y.toString() + "x" + x.toString())

            val density = displayMatrix.density
            val densityDpi = displayMatrix.densityDpi

            val d = ChildView(activity, R.string.deviceinfo_display_density, density.toString())
            val dd = ChildView(activity, R.string.deviceinfo_display_densityDpi, densityDpi.toString())

            val scaledDensity = displayMatrix.scaledDensity
            val s =
                ChildView(activity, R.string.deviceinfo_display_scaledDensity, scaledDensity.toString())

            activity!!.runOnUiThread { displayRoot.addViews(r, d, dd, s) }
        }
        t6.start()

        Thread {
            while (t1.isAlive || t2.isAlive || t3.isAlive || t4.isAlive || t5.isAlive || t6.isAlive) {
                Thread.sleep(1)
            }
            Timer().schedule(object : TimerTask() {
                override fun run() {
                    activity!!.runOnUiThread { dialog.cancel() }
                }
            }, 1000)
        }.start()
    }

}